src/requestor.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {AppAuthError} from './errors';
+import {StringMap} from './types';
+
+export interface RequestorBaseSettings {
+  /** URL to request */
+  url: string;
+  /** HTTP method, as uppercase string */
+  method: string;
```

Comment:
Rather than making this a `string` type. Constrain it using something like:

```
type RequestMethod = 'GET'|'POST'|'HEAD'|'PUT'|'DELETE'
```

Reply:
It just didn't seem necessary - the base interface is only there to reduce documentation/property duplication for GET/POST variants currently used. The interface for xhr is constrained such that it won't allow method: 'DELETE' without a change.